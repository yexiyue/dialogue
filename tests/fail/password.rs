#![allow(unused)]
use dialogue_macro::Asker;

#[derive(Asker, Debug)]
struct User {
    #[password(
        prompt = "Enter your password:",
        confirm = "Enter your password again:",
        mismatch = "Your passwords do not match"
    )]
    password2: String,
    #[password()]
    password: Option<bool>,
}

fn main() {
    let user = User::asker()
        .password2()
        .password("Enter your password:")
        .finish();
    println!("{:?}", user);
}
