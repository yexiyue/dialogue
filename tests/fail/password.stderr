error: expected `prompt` , `confirmation` , `mismatch` , `validate_with`
 --> tests/fail/password.rs:8:9
  |
8 |         confirm = "Enter your password again:",
  |         ^^^^^^^

error[E0599]: no function or associated item named `asker` found for struct `User` in the current scope
  --> tests/fail/password.rs:17:22
   |
5  | struct User {
   | ----------- function or associated item `asker` not found for this struct
...
17 |     let user = User::asker()
   |                      ^^^^^ function or associated item not found in `User`
